File: app/src/main/java/it/liceoarzignano/bold/home/HomeCard.java
package it.liceoarzignano.bold.home;

import android.view.View;

import java.util.List;

public class HomeCard {
    private final int mSize;
    private final String mName;
    private final List<String> mTitle;
    private final List<String> mContent;
    private final HomeCardClickListener mClickListner;

    HomeCard(int size, String name, List<String> title, List<String> content,
             HomeCardClickListener clickListener) {
        this.mSize = size;
        this.mName = name;
        this.mTitle = title;
        this.mContent = content;
        this.mClickListner = clickListener;
    }

    public int getSize() {
        return mSize;
    }

    public String getName() {
        return mName;
    }

    public List<String> getTitle() {
        return mTitle;
    }

    public List<String> getContent() {
        return mContent;
    }

    void doClickAction(View view) {
        if (mClickListner == null) {
            return;
        }

        mClickListner.onClick(view);
    }

    public interface HomeCardClickListener {
        void onClick(View view);
    }
}

File: app/src/main/kotlin/it/liceoarzignano/bold/home/HomeCard.kt
package it.liceoarzignano.bold.home

class HomeCard(val type: CardType) {

    var title = ""
    var content = ""
    var action = ""
    var listener: Listener? = null

    fun onCardClick() = listener?.onCardClick()
    fun onActionClick() = listener?.onActionClick()

    interface Listener {
        fun onCardClick()
        fun onActionClick()
    }

    enum class CardType {
        EVENTS,
        NEWS,
        MARKS,
        SUGGESTIONS
    }
}
